addon_toolkit/credentials.py

```diff
@@ -27,6 +27,36 @@ class AccessKeySecretKeyCredentials(Credentials):
     secret_key: str
 
 
+@dataclasses.dataclass(frozen=True, slots=True)
+class OAuth1Credentials(Credentials):
+    oauth_token: str
+    oauth_token_secret: str
+
+    def iter_headers(self) -> typing.Iterator[tuple[str, str]]:
+        """
+        This is Zotero specific as other OAuth1.0a clients require request signing,
+        as per current architecture, we cannot it here.
+        """
+
+        yield "Authorization", f"Bearer {self.oauth_token_secret}"
+        # TODO: implement request signing for OAuth1.0a services that require it
```

Comment:
gah, i missed that zotero uses the secret as bearer token instead of signing the request... feels wrong, but maybe it's true that `https` makes it safe, as argued in [their example client code](https://www.zotero.org/support/dev/web_api/v3/oauth#example_php):
```
/** OAuth support for all api requests may be added in the future
* but for now secure https provides similar benefits anyway
*/
```

(this speaks to the need for modular/swappable request-auth logic (instead of `iter_headers`) but sounds like that'll be part of s3 work -- this TODO seems fine for now)

Reply:
We should be safe because `https` encypts entire `http` payload (including headers, query params, path, body, etc.)